### DISCUSSION TITLE
Out of Memory in HeteroGraph Explanation

### DISCUSSION THREAD

**Discussion by Mellandd** [pyg-team/pytorch_geometric#7718] [Q&A]
Hello,
I am trying to use the interpretability package with a model created for a heterogeneous graph. The problem comes in that when trying to generate the interpretation there is not enough memory. 

It is a model with 6 layers and hidden channel of 256, based on GraphSAGE for nodes of 2 categories. The dataset is:

`HeteroData(
  a={ x=[16810, 768] },

  b={ x=[4619, 53] },

  (a, related_to, b)={ edge_index=[2, 721540] },

  (a, is_a, a)={ edge_index=[2, 42142] },

  (b, rev_related_to, a)={ edge_index=[2, 721540] }
)`

The GPU has 8 GB of memory. Making the model smaller right now is not an option. Is there anything I can do to be able to use the interpretability package? 

- I am using CaptumExplainer because as far as I have seen it is the only one that works with heterogeneous graphs. Are there any others or are there any plans to have compatibility soon?
- According to what I have seen, CaptumExplainer has not implemented the use of batches yet. Is there any news about this?
- I have 2 GPUs in the computer. Is it possible to use both in order not to have memory problems in this task?

Thank you all!

**Comment by wsad1** [comment]
You could save memory by only explaining a subgraph. 
For example lets say you want to explain nodes `[1,10]`. You could pass this to [NeighborLoader](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch_geometric.loader.NeighborLoader) and explain only the subgraphs returned by `NeighborLoader`.

**Reply by rusty1s** [reply]
Yes, it seems that `CaptumExplainer` has a higher memory footprint than other explainers, so there is not much we can do about this right now besides reducing the input data to a subgraph.
